### PR TITLE
feat: override compiler's `outputFileSystem` to a fully featured `fs`

### PR DESCRIPTION
**Pull request by andrefgneves**

### CODE REVIEW

File: lib/standardFs.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+'use strict';
+
+const fs = require('fs');
+const mkdirp = require('mkdirp');
```
Comment (satazor) [inline lib/standardFs.js]: Shouldn’t this be added as a dependency?

File: index.js [hunk 2]
```diff
@@ -5,6 +5,7 @@ const webpack = require('webpack');
 const pDefer = require('p-defer');
 
 const observeWebpackCompiler = require('./lib/observeWebpackCompiler');
+const standardFs = require('./lib/standardFs');
```
Comment (satazor) [inline index.js]: Maybe change the module to nodeFs

Comment (satazor) [reply index.js]: Also should we add that we are doing this in the README?

File: index.js [hunk 3]
```diff
@@ -26,6 +27,8 @@ function compiler(webpackArg) {
 
     const { eventEmitter, state } = observeWebpackCompiler(webpackCompiler);
 
+    webpackCompiler.outputFileSystem = standardFs();
```
Comment (satazor) [inline index.js]: Could you add a test for this?

File: lib/standardFs.js (hunk 1 above)
Comment (andrefgneves) [reply lib/standardFs.js]: Done

File: index.js (hunk 2 above)
Comment (andrefgneves) [reply index.js]: Done

File: index.js (hunk 3 above)
Comment (andrefgneves) [reply index.js]: Done

File: README.md [hunk 4]
```diff
@@ -182,3 +183,5 @@ You may also want to look at:
 ## License
 
 [MIT License](http://opensource.org/licenses/MIT)
+
```
Comment (satazor) [inline README.md]: Remove the extra lines added.

File: test/misc.spec.js [hunk 5]
```diff
@@ -23,6 +24,12 @@ it('should prevent direct access to webpack compiler\'s main methods', () => {
     expect(() => compiler.webpackCompiler.watch).toThrow(/\bpublic API\b/);
 });
 
+it('should override the webpack compiler\'s outputFileSystem to a fully featured node fs', () => {
+    const compiler = createCompiler(configBasic);
+
+    expect(compiler.webpackCompiler.outputFileSystem).toMatchObject(nodeFs());
```
Comment (satazor) [inline test/misc.spec.js]: I think that the test should assert that the result of nodeFs() matches require(‘fs’) and has mkdirp.
The reasoning is that this test won’t fail even if nodeFs() returned an empty object.

File: README.md (hunk 4 above)
Comment (andrefgneves) [reply README.md]: 👍

File: test/misc.spec.js (hunk 5 above)
Comment (andrefgneves) [reply test/misc.spec.js]: 👍

**Comment by satazor** [comment]
Cool, you may release a new version and update the other MR.

**Pull request merged**